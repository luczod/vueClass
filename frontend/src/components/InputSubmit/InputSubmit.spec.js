import { describe, expect, it } from 'vitest';
import Inputsubmit from './index.vue';
import { mount } from '@vue/test-utils';

describe('<InputSubmit />', () => {
  it('should be defined', () => {
    const wrapper = mount(Inputsubmit);
    expect(wrapper).toBeDefined();
  });

  it('asserts correct props are passed', () => {
    const wrapper = mount(Inputsubmit, {
      props: {
        msg: 'Cadastrar',
        blocked: false,
      },
    });

    expect(wrapper.vm.msg).toBe('Cadastrar');
    expect(wrapper.vm.blocked).toEqual(false);
  });

  it('should match rendered', () => {
    const wrapper = mount(Inputsubmit);
    expect(wrapper.element).toMatchSnapshot();
  });
});
