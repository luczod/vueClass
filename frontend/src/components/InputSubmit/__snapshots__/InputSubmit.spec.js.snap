// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<InputSubmit /> > should match rendered 1`] = `
<input
  class="btn"
  data-v-8bf9b1d7=""
  type="submit"
/>
`;
